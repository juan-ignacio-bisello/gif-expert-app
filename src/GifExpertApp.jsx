import { useState } from "react";
import { AddCategory } from './components/AddCategory'

export const GifExpertApp = () => {

  const [ categories , setCategorias ] = useState([ 'One Punch', 'Dragon Ball' ]);

  const onAddCategory = () => {
    
    setCategorias([ 'Black Clover', ...categories ]);
    //Otra forma
    // setCategorias( cat => [...cat, 'Black Clover']);

  }


  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>
        
      <AddCategory />

      {/* Input */}
      <button onClick={ onAddCategory }>Agregar</button>

      {/* Listado de Gif */}
      <ol>
        {
        categories.map( category => {
            return <li key={ category }>{ category }</li>
          })
        }
      </ol>
    </>
   
  )
}
