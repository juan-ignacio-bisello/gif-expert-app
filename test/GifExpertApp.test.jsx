import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp";

describe('Puebas en GifExpertApp', () => { 
    
    const inputValue = 'Saitama';

    test('debe', () => { 
        
        //HACER UN SNAPSHOT
        const { container } = render( <GifExpertApp />);
        expect( container ).toMatchSnapshot();

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );

        expect( container.getElementsByClassName('card-grid').length ).toBe( 2 )
     });

    test('no debe agregar una nueva categoria si la categoria ya existe', () => {
        
        // Renderizamos el sujeto de pruebas y obtenemos el container
        const { container } = render( <GifExpertApp /> );
 
        // Obtenemos el input y el form
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
 
        // Disparo los eventos para a√±adir las categorias
 
        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form ); // Se agrega 'Saitama'
 
        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form ); // Se agrega 'Saitama' por segunda vez
 
        // Utilizamos el container para buscar las categorias agregadas y verificamos que solo se haya agregado la categoria una vez
        expect( container.getElementsByClassName('card-grid').length ).toBe( 2 );
 
    });
 })