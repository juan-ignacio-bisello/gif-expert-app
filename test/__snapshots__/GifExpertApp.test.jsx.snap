// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en GifExpertApp debe 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
